integration-cli/docker_cli_ps_test.go

```diff
@@ -787,3 +787,30 @@ func (s *DockerSuite) TestPsShowMounts(c *check.C) {
 	out, _ = dockerCmd(c, "ps", "--format", "{{.Names}} {{.Mounts}}", "--filter", "volume="+prefix+slash+"this-path-was-never-mounted")
 	c.Assert(strings.TrimSpace(string(out)), checker.HasLen, 0)
 }
+
+func (s *DockerSuite) TestPsListContainersFilterNetwork(c *check.C) {
+	// create a container
+	out, _ := dockerCmd(c, "run", "-d", "--net=bridge", "--name=onbridgenetwork", "busybox", "top")
+	out, _ = dockerCmd(c, "run", "-d", "--net=none", "--name=onnonenetwork", "busybox", "top")
+
+	// Filter docker ps on network bridge
+	out, _ = dockerCmd(c, "ps", "--filter", "network=bridge")
+	containerOut := strings.TrimSpace(string(out))
+
+	lines := strings.Split(containerOut, "\n")
+
+	// skip header
+	lines = lines[1:]
+
+	// ps output should have only one container
+	c.Assert(lines, checker.HasLen, 1)
+
+	// Making sure onbridgenetwork is on the output
+	c.Assert(lines[0], checker.Contains, "onbridgenetwork", check.Commentf("Missing the container on network\n"))
+
+	dockerCmd(c, "stop", "onbridgenetwork")
```

Comment:
You don't have to cleanup containers at the end of the test; it's done automatically between tests. See https://github.com/docker/docker/blob/4a7bd7eaef0038b006cf23adface84081e5aab3b/integration-cli/check_test.go#L33-L39